### ISSUE TITLE
Rewrite upgrade documentation to remove repetition 

### ISSUE THREAD

**Issue by jpeach** [projectcontour/contour#2365]
As currently constructed, the [upgrade](https://projectcontour.io/resources/upgrading/) document needs manual updating every release. Most of the upgrade instructions should be the same (i.e. change image versions). We should rewrite this to be more generically useful and descriptive (e.g. how do you validate an upgrade, or verify the current version), remove repetition, and not require manual edits on every release.

**Comment by stevesloka** [comment]
I agree this is repetitive. Might also be helpful to find a way to split apart the document or just add to the release notes since they sort of fit together (the release notes + upgrade guides)

**Comment by skriss** [comment]
Closing this out as a duplicate of #3223, which has a bit more discussion on it.
